blueprint/core/blueprint_HotReloadingHarness.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+  ==============================================================================
+
+    blueprint_HotReloadingHarness.cpp
+    Created: 21 Nov 2020 11:27:37am
+
+  ==============================================================================
+*/
+
+#pragma once
+
+#include "blueprint_HotReloadingHarness.h"
+
+
+namespace blueprint
+{
+
+    //==============================================================================
+    HotReloadingHarness::HotReloadingHarness(ReactApplicationRoot& appRoot, std::shared_ptr<EcmascriptEngine> ee)
+    {
+        JUCE_ASSERT_MESSAGE_THREAD
+
+        fileWatcher = std::make_unique<FileWatcher>([this, &appRoot, ee]() {
+            appRoot.reset();
+            appRoot.bindNativeRenderingHooks();
+
+            // TODO: I could imagine support 4 hooks here: beforeAll, beforeEach, afterEach, beforeAll,
+            // where the middle two callbacks actually get specific references to _which_ bundle
+            if (beforeBundleEval)
+                beforeBundleEval(ee, juce::File());
```

Comment:
So in the default implementation we're just passing a default constructed `juce::File` to the callbacks? In which case, given we aren't making `HotReloadingHarness` a virtual class to be overridden by special implementations, should we just do away with the `juce::File` argument in those callbacks?

Reply:
Hacks again hahah sorry I should have explained this better... see my comment above about where I think we should take this, and then I'll come back and do it properly